src/vga/vga_buffer.rs

```diff
@@ -13,18 +19,20 @@ pub struct VGADevice<'a> {
 }
 
 impl Clearable for VGADevice<'_> {
-    fn clear(&mut self, color: &VGAColor) {
+    fn clear(&mut self, color: VGAColor<u8>) {
         for x in 0..self.frame_buffer_info.horizontal_resolution {
             for y in 0..self.frame_buffer_info.vertical_resolution {
-                self.draw_point(x, y, color);
+                self.draw_point(x as u16, y as u16, color);
             }
         }
     }
 }
 
 impl PlainDrawable for VGADevice<'_> {
-    fn draw_point(&mut self, x: usize, y: usize, color: &VGAColor) {
-        let index = y * self.bytes_per_row + x * self.frame_buffer_info.bytes_per_pixel;
+    fn draw_point(&mut self, x: u16, y: u16, color: VGAColor<u8>) {
+        let _x = x as usize;
```

Comment:
just a tip in case you didn't know: you can re-assign variables if you want, `let x = x as usize`. It's standard practice to do this in cases as such (also prefixing variables with `_` in rust is usually meant to signal that that variable is ignored)

Reply:
Oh, that's actually very good to know - will use that